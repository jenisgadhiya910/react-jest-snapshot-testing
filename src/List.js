import React, { Component } from 'react';

export default class List extends Component {
    render() {

        const listItems = this.props.items ? this.props.items.map((item, index) => (
            <li key={index}>{item}</li>
        )) : [];

        return (
            <ul>
                {listItems}
            </ul>
        )
    }
}