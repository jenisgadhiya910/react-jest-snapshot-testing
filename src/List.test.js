import React from 'react';
import renderer from 'react-test-renderer';
import List from './List';

describe('List', () => {

  it('matches snapshot when empty list', () => {
    const tree = renderer
      .create(<List/>);
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot when items passed in', () => {
    const items = ['abc', 'def', 'ghi'];
    const tree = renderer
      .create(<List items={items}/>);
    expect(tree).toMatchSnapshot();
  });

});