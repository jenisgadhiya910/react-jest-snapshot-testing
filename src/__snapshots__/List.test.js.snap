// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List matches snapshot when empty list 1`] = `<ul />`;

exports[`List matches snapshot when items passed in 1`] = `
<ul>
  <li>
    abc
  </li>
  <li>
    def
  </li>
  <li>
    ghi
  </li>
</ul>
`;
